content/en/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice.md

```diff
@@ -13,56 +13,183 @@ variables and then consume them using MicroProfile Config.
+
+2. **A running Kubernetes cluster**
+
+   Set up a Kubernetes cluster according to your platform:
+
+   - For **Mac** users, you can use [Docker Desktop](https://www.docker.com/products/docker-desktop) to run a Kubernetes cluster locally.
+
+   - For **Windows** users, Docker Desktop also supports Kubernetes on Windows.
+
+   - For **Linux** users, you can use [Minikube](https://minikube.sigs.k8s.io/docs/start/) or set up a Kubernetes cluster on a cloud provider like [Google Kubernetes Engine (GKE)](https://cloud.google.com/kubernetes-engine) or [Amazon Elastic Kubernetes Service (EKS)](https://aws.amazon.com/eks/).
+
+3. **Java development environment set up on your local machine**
+
+   Ensure you have Java installed. You can check by running `java -version` in your terminal. If not installed, follow these steps:
+
+   - **Mac**:
+
+     ```bash
+     brew install openjdk@11
+     ```
+
+   - **Windows**:
+     Download and install [AdoptOpenJDK](https://adoptopenjdk.net/) for Java 11.
+
+   - **Linux**:
+     Use your package manager to install OpenJDK 11. For example, on Ubuntu:
+
+     ```bash
+     sudo apt-get update
+     sudo apt-get install openjdk-11-jdk
+     ```
+
+Now that you have the prerequisites in place, let's proceed with the tutorial.
+
+## Creating Kubernetes ConfigMaps & Secrets
 
 There are several ways to set environment variables for a Docker container in Kubernetes,
-including: Dockerfile, kubernetes.yml, Kubernetes ConfigMaps, and Kubernetes Secrets.  In the
-tutorial, you will learn how to use the latter two for setting your environment variables whose
-values will be injected into your microservices.  One of the benefits for using ConfigMaps and
-Secrets is that they can be re-used across multiple containers, including being assigned to
-different environment variables for the different containers.
-
-ConfigMaps are API Objects that store non-confidential key-value pairs.  In the Interactive
-Tutorial you will learn how to use a ConfigMap to store the application's name.  For more
+including: Dockerfile, kubernetes.yml, Kubernetes ConfigMaps, and Kubernetes Secrets.  In this section, we'll explore how to use ConfigMaps and Secrets to manage your microservice's environment variables.
+
+### ConfigMaps: Storing Non-Sensitive Data
+
+ConfigMaps are perfect for storing non-sensitive configuration data that can be shared across multiple containers.
+
+**Step 1: Create a ConfigMap**
+
+To get started, follow these steps:
+
+1. Open a text editor of your choice.
+
+2. Create a new file with a `.yaml` extension, for example, `myapp-configmap.yaml`.
+
+3. Copy and paste the following content into the `myapp-configmap.yaml` file:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: myapp-config
+data:
+  app.name: MyApp
+  api.endpoint: https://api.example.com
+```    
+ConfigMaps are API Objects that store non-confidential key-value pairs.  For more
```

Comment:
This sentence feels misplaced and should come before the tutorial IMO. WDYT?

Reply:
Hi @divya-mohan0209, made the changes.